enumerables.rb

```diff
@@ -0,0 +1,142 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = is_a?(Array) ? self : to_a
+    i = 0
+    # puts arr
+    while i < arr.length
+      yield arr[i]
+      i += 1
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = is_a?(Array) ? self : to_a
+    i = 0
+    # puts arr
+    while i < arr.length
+      yield arr[i], i
+      i += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    arr = to_a
+    new_array = []
+    arr.my_each do |item|
+      new_array.push(item) if yield item
+    end
+    new_array
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+
+    elsif arg.nil?
+      my_each { |n| return false if n.nil? || n == false }
+    elsif arg.is_a? Class
+      my_each { |n| return false unless n.is_a? arg || n.instance_of?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      my_each { |n| return false unless arg.match(n) }
+    else
+      my_each { |n| return false if n != arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+    elsif arg.nil?
+      my_each { |n| return true if n.nil? || n == false }
+    elsif arg.is_a? Class
+      my_each { |n| return true if n.is_a? arg || n.instance_of?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      my_each { |n| return true if arg.match(n) }
+    else
+      my_each { |n| return true if n == arg }
+    end
+    false
+  end
+
```

Comment:
### #my_any? when no block or argument is given returns false if at least one of the collection is not true